src/org/daisy/dotify/formatter/impl/page/PageSequenceBuilder2.java

```diff
@@ -714,6 +725,10 @@ private BlockContext addRows(List<RowGroup> head, PageImpl p, BlockContext block
                 }
             }
         }
+
+        // After we have written the row group we are no longer at the top of the page so the mutable
+        // value topOfPage needs to change so we will rebuild an object in order to change this context
+        // to false.
         return new BlockContext.Builder(blockContext).topOfPage(false).build();
```

Comment:
Line should be moved (basically revert 31534105c933de51713ac6bf8b2f2e912eb69a92, see other comments).